### ISSUE TITLE
Pyinstaller + PyQt 5.11.2 OSX : Qt WebEngine ICU data not found

### ISSUE THREAD

**Issue by rllola** [pyinstaller/pyinstaller#4030]
Hello,

I am currently trying to build my application for OSX with PyQt5. I am keep getting an error when trying to run  that it cannot find `Qt WebEngine ICU data`.

Here the full log : 
```
Qt WebEngine ICU data not found at . Trying parent directory...

Qt WebEngine ICU data not found at /Users/kek/Desktop/dist/ZeronetBrowser.app/Ressources/PyQt5/Qt/lib/QtWebEngineCore.framework/Resources. Trying application directory...

Qt WebEngine ICU data not found at /Users/kek/Desktop/dist/ZeronetBrowser.app/Contents/MacOS. Trying fallback directory... The application MAY NOT work.

Path override failed for key base::DIR_QT_LIBRARY_DATA and path '/Users/kek/.ZeronetBrowser'

Installed Qt WebEngine locales directory not found at location /qtwebengine_locales. Trying application directory...

Qt WebEngine locales directory not found at location /Users/kek/Desktop/dist/ZeronetBrowser.app/Contents/MacOS/qtwebengine_locales. Trying fallback directory... Translations MAY NOT not be correct.

Path override failed for key ui::DIR_LOCALES and path '/Users/kek/.ZeronetBrowser'
```

And I have the in the `/qtwebengine_locales` in here `ZeronetBrowser.app/Contents/Resources/PyQt5/Qt/lib/QtWebEngineCore.framework/Resources/`

I did notice that the path build by PyInstaller doesn't match the one recommended in qt documentation : https://doc.qt.io/qt-5/qtwebengine-deploying.html

Any reason why ? Does someone knows what I am doing wrong ?

Project : https://github.com/rllola/ZeronetBrowser
Release tested by my friend : https://github.com/rllola/ZeronetBrowser/releases/tag/v0.1.10
Python 2.7
Pyinstaller 3.4

**Comment by bjones1** [comment]
This *might* be fixed in develop. However, it's tested on Python 3, not Python 2.

**Comment by rllola** [comment]
So it didn't changed anything still the same error.

**Comment by bjones1** [comment]
Another guess: use the `--windowed` option when building. See #3661. #3506.

**Comment by rllola** [comment]
> Another guess: use the `--windowed` option when building. See #3661. #3506.

Yes, we are doing that already in our branch `mac-bundle` : https://github.com/rllola/ZeronetBrowser/tree/mac-bundle

I did modify the `qt.conf` file tho. Could it has a wrong path ? 

```
[Paths]
Prefix = ../Resources/PyQt5/Qt
Data = lib/QtWebEngineCore.framework/Resources
Translations = lib/QtWebEngineCore.framework/Resources
```
I had a typo before that I fixed. Should I just not have one ?

**Comment by bjones1** [comment]
CI tests run (and pass) on the default `qt.conf` provided in the Mac wheel from PyPI. Other than that, I can't help -- I don't own a Mac, so my experience is very limited. If you do figure anything out, please submit a PR; @glennra or @tallforasmurf might have some ideas.

**Comment by rllola** [comment]
Where can I see the CI tests ? 

We will figure this out thank you.

For @glennra and @tallforasmurf I suspect that something is happening here : https://github.com/pyinstaller/pyinstaller/blame/develop/PyInstaller/hooks/hook-PyQt5.QtWebEngineWidgets.py#L22-L32

What it does is different from is described here : 
https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/hooks/qt.py#L309-L325

**Comment by rllola** [comment]
> CI tests run (and pass) on the default qt.conf provided in the Mac wheel from PyPI.

Oh wait... I don't have that. I am building everything from scratch because I am on python 2.7 (not because I choose it).

Can I get a copy of it ?

**Comment by bjones1** [comment]
The CI test in question is https://github.com/pyinstaller/pyinstaller/blob/develop/tests/functional/test_libraries.py#L258.

You can copy the `qt.conf` from the Mac PyPI wheel. I don't have it handy, however.

**Comment by rllola** [comment]
Cool thank you.

One other thing. I tried to build with the development version of PyInstaller in travis (I don't have MacsOS neither). And now my package is 1GB !! Something went terribly wrong...

https://github.com/rllola/ZeronetBrowser/releases/tag/v0.1.11

**Comment by bjones1** [comment]
I haven't seen that. I'd suggest using a venv and install only packages you require, since PyInstaller will only include those.

**Comment by rllola** [comment]
Ok some good news !

Travis test seems to pass : https://travis-ci.org/rllola/WebBrowser-PyQt5/builds/488241168

I still have warnings tho but at least it seems to work. I will create a github release and see if it is working for people on their computer.

I had to specify in `qt.conf` the `Data` Path. See it here : https://github.com/rllola/WebBrowser-PyQt5/blob/master/qt.conf

Also I repeat that the dev version of Pyinstaller break this.

**Comment by bjones1** [comment]
Given that CI passes on Python 3 + PyQt5, I suspect this arises from some Python 2-specific problems. If you can track down the error, let me know. However, given Python 2's upcoming EOL, I don't see this as a priority.

**Comment by rllola** [comment]
> However, given Python 2's upcoming EOL, I don't see this as a priority.
I 100% agree.

I did keep digging into it. It might not be linked to python2 or python3 entirely but more on how I built PyQt5 for python2 I had to build from scratch and the `site-package/PyQt5` doesn't have the same folders than when I installed it doing `pip3 install PyQt5` for python3.

I am not expecting an answer. I am just posting what I have found for archive.
